templates/repo/diff/box.tmpl

```diff
@@ -65,6 +69,12 @@
 							{{end}}
 						</div>
 						<span class="file mono">{{if $file.IsRenamed}}{{$file.OldName}} &rarr; {{end}}{{$file.Name}}{{if .IsLFSFile}} ({{$.i18n.Tr "repo.stored_lfs"}}){{end}}</span>
+						{{if $file.IsGenerated}}
+							<span class="ui label ml-3 mr-3">{{$.i18n.Tr "repo.diff.generated"}}</span>
+						{{end}}
+						{{if $file.IsVendored}}
```

Comment:
If I see that correctly, couldn't that potentially result in a file where both the `isGenerated` and the `isVendored` are displayed, assuming a vendored and generated file would exist?
Would it be better to make this an `else if` then or can this case be ignored?

Reply:
Yes vendored and generated could be both displayed. They appear to be orthogonal statuses and we calculate both so I don't see a good reason not to display both.

Reply:
Okay, yeah, I can see that.